conda_env/specs/binstar.py

```diff
@@ -0,0 +1,111 @@
+import re
+from conda.resolve import normalized_version
+from .. import env
+from ..exceptions import EnvironmentFileDoesNotExist, EnvironmentFileNotDownloaded, CondaEnvException
+try:
+    from binstar_client import errors
+    from binstar_client.utils import get_binstar
+except ImportError:
+    raise CondaEnvException("Binstar not installed")
+
+ENVIRONMENT_TYPE = 'env'
+
+
+class BinstarSpec(object):
+    """
+    spec = BinstarSpec('darth/deathstar')
+    spec.can_handle() # => True / False
+    spec.environment # => YAML string
+    spec.msg # => Error messages
+    :raises: EnvironmentFileDoesNotExist, EnvironmentFileNotDownloaded
+    """
+
+    _environment = None
+    _username = None
+    _packagename = None
+    _package = None
+    _file_data = None
+    msg = None
+
+    def __init__(self, filename=None, name=None, **kwargs):
+        self.filename = filename
+        self.name = name
+        self.binstar = get_binstar()
+        self.quiet = False
+        # self.info("Successfully fetched {} from Binstar.org".format(self.handle))
+
+    def can_handle(self):
+        """
+        Validates loader can process environment definition.
+        :return: True or False
+        """
+        # TODO: log information about trying to find the package in binstar.org
+        return self.valid_name() and self.package is not None and self.valid_package()
+
+    def valid_name(self):
+        """
+        Validates name
+        :return: True or False
+        """
+        if re.match("^(.+)/(.+)$", str(self.name)) is not None:
+            return True
+        elif self.name is None:
+            self.msg = "Can't process without a name"
+        else:
+            self.msg = "Invalid name, try the format: user/package"
+        return False
+
+    def valid_package(self):
+        """
+        Returns True if package has an environment file
+        :return: True or False
+        """
+        return len(self.file_data) > 0
+
+    @property
+    def file_data(self):
+        if self._file_data is None:
+            self._file_data = [data for data in self.package['files'] if data['type'] == ENVIRONMENT_TYPE]
+        return self._file_data
+
+    @property
+    def environment(self):
+        """
+        :raises: EnvironmentFileNotDownloaded
+        """
+        if self._environment is None:
+            versions = {normalized_version(d['version']): d['version'] for d in self.file_data}
+            latest_version = versions[max(versions)]
+            file_data = [data for data in self.package['files'] if data['version'] == latest_version]
+            req = self.binstar.download(self.username, self.packagename, latest_version, file_data[0]['basename'])
```

Comment:
Is there not a way to download the latest package for a given package?  That seems like logic that should be on the server side, not the client.